### PR TITLE
Put test into unnamed namespace as much as possible

### LINKED ISSUE #5856: test: put test into unnamed namespace as much as possible
*Description*:
There is no reason tests are not in unnamed namespace unless needed by `FRIEND_TEST`, they don't need external linkage. It also saves about 10% of compiling / linking time for the test.cc and binary based on my quick experiment with GCC.

### PR DESCRIPTION
**Pull request by solodynamo**
Description: Puts test into unnamed namespace as much as possible.
Risk Level: Low
Testing: 
Docs Changes: No
Release Notes: NA
Issues: Fixes #5856

### CODE REVIEW

**Comment by lizan** [comment]
@solodynamo Thanks for taking this up! Looks CI is failing, not all of tests can be put into unnamed namespaces. Also it would be nice to put some check into the format check for this.

/wait

**Comment by solodynamo** [comment]
@lizan Sure will look into it.

**Comment by solodynamo** [comment]
@htuch At current all .cc files in testing directory has been namespaced but as it sits I have to remove anonymous namespaces from some files as CI tests are breaking.

**Comment by lizan** [comment]
Can you merge master to pick a flaky test fix?

/retest

File: test/common/network/listen_socket_impl_test.cc [hunk 1]
```diff
@@ -15,6 +15,7 @@ using testing::Return;
 namespace Envoy {
 namespace Network {
 
+namespace {
```
Comment (alyssawilk) [inline test/common/network/listen_socket_impl_test.cc]: nit: mind fixing up whitespace by moving this up a line?

File: test/common/access_log/access_log_formatter_fuzz_test.cc [hunk 2]
```diff
@@ -21,6 +22,6 @@ DEFINE_PROTO_FUZZER(const test::common::access_log::TestCase& input) {
     ENVOY_LOG_MISC(debug, "EnvoyException: {}", e.what());
   }
 }
-
+} // namespace
```
Comment (alyssawilk) [inline test/common/access_log/access_log_formatter_fuzz_test.cc]: I hope it's not too much of a pain but I do think it's a bit more readable to have the blank line after a namespace declaration and before the end-namespace brace.  Do you think you could add those back?

Comment (solodynamo) [reply test/common/access_log/access_log_formatter_fuzz_test.cc]: Yeah sure, will do

**Comment by solodynamo** [comment]
`Unable to load package for '//bazel:repositories.bzl` Error in CI doesn't seem to be related to my PR. Pls, look into this @lizan @htuch @alyssawilk

**Comment by mattklein123** [comment]
Thanks this is great. Merging to avoid conflicts.

**Pull request merged**